RIOT-OS/RIOT
# drivers/mrf24j40: add support for IEEE 802.15.4 Radio HAL

### Contribution description
Since https://github.com/RIOT-OS/RIOT/pull/18383 already got merged, it's time to add the IEEE 802.15.4 Radio HAL port for MRF24J40 radios.
This one was supported by LWIP, which requires https://github.com/RIOT-OS/RIOT/pull/18465 to be merged first.

This PR also adds the Arduino Pin config for nucleo64 boards and fixes some tiny issues in the CCA functions of the Radio HAL and tests.



Strictly speaking this PR changes the API of `mrf24j40_init`, but IMO adding a deprecation at this level does not make sense.
It also removes the old `netdev` implementation.

### Testing procedure
Connect a MRF24J40 to the Arduino Pins, try `tests/ieee802154_*` tests, the LWIP tests and `examples/gnrc_networking`. Everything should work as before.



### Issues/PRs references
~Depends on https://github.com/RIOT-OS/RIOT/pull/18465~
